src/libraries/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.Aead.cs

```diff
@@ -16,6 +16,17 @@ internal static partial class Interop
 {
     internal static partial class AppleCrypto
     {
+        private static byte NullSentinel;
+
+        // CryptoKit doesn't do well with a null pointer for the buffer data,
```

Comment:
We were passing in NULL before. What changed that made it not permitted?

Reply:
The Foundation.Data type turns a zero-length span to point to the inline storage. This PR has us using the UnsafeBufferPointer type, which apparently Swift doesn't do well with on x64 (works fine on ARM64 for some reason).